### Present Value of Liabilities and Funding Ratio

$$ PV(L) = \sum_{i=1}^{k} B(t_i) L_i$$

When:

$$B(t) = \frac{1}{(1+r)^t}$$

In [1]:
import numpy as np
import pandas as pd
import risk_toolkit as rk
import ipywidgets as widgets
from IPython.display import display
%load_ext autoreload
%autoreload 2
%reload_ext autoreload
%matplotlib inline

In [2]:
rk.discount(10, 0.03)

0.7440939148967249

In [3]:
rk.discount(10, 0.03)

0.7440939148967249

In [4]:
liabilities = pd.Series(data=[1, 1.5, 2, 2.5], index=[3, 3.5, 4, 4.5])

In [5]:
rk.present_value(liabilities, 0.03)

6.233320315080044

Interpretation: I owe USD 7 million (liabilities) in the future, but the present value of that is USD 6.23 million

In [6]:
rk.funding_ratio(5, liabilities, 0.03)

0.8021407126958778

Interpretation: We are under-funded. The USD 5 million in assets cannot meet all future liabilities

In [21]:
def show_funding_ratio(assets, interest_rate):
    
    fr = rk.funding_ratio(assets, liabilities, interest_rate)
    
    print(f'Funding ratio: {fr*100:.2f}')

In [26]:
controls = widgets.interactive(show_funding_ratio,
                               assets = widgets.IntSlider(min=1, max=10, step=1, value=5, description='Assets'),
                               interest_rate = widgets.FloatSlider(min=0.0, max=0.20, step=0.01, value=0.03, description='Interest rate')
                              )

In [27]:
display(controls)

interactive(children=(IntSlider(value=5, description='Assets', max=10, min=1), FloatSlider(value=0.03, descrip…